sonar-php-plugin/src/main/java/org/sonar/plugins/php/AnalysisScanner.java

```diff
@@ -218,6 +225,27 @@ protected boolean fileCanBeSkipped(InputFile file) {
     return super.fileCanBeSkipped(file) || (file.type() == InputFile.Type.TEST && !hasTestFileChecks);
   }
 
+  private List<PhpIssue> filterIssuesByWarningSuppressor(InputFile inputFile, List<PhpIssue> issues) {
+    return issues.stream()
+      .filter(issue -> !isSuppressed(inputFile, issue))
+      .collect(Collectors.toList());
+  }
+
+  private boolean isSuppressed(InputFile inputFile, PhpIssue issue) {
+    RuleKey ruleKey = checks.ruleKeyFor(issue.check());
+    if (ruleKey != null) {
+      if (issue instanceof LineIssue) {
+        LineIssue lineIssue = (LineIssue) issue;
+        return suppressWarningFilter != null && !suppressWarningFilter.accept(inputFile.uri().toString(), ruleKey.rule(), lineIssue.line());
+      } else if (issue instanceof PreciseIssue) {
+        PreciseIssue preciseIssue = (PreciseIssue) issue;
+        return suppressWarningFilter != null && !suppressWarningFilter.accept(inputFile.uri().toString(), ruleKey.rule(), preciseIssue.primaryLocation().startLine());
```

Comment:
To make it more readable i would suggest negating a lot of the boolean logic. Like `isIncluded` instead of `isSuppressed`